sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/TableRowToStorageApiProto.java

```diff
@@ -681,7 +681,7 @@ private static void fieldDescriptorFromTableField(
               new BigDecimal((String) value));
         } else if (value instanceof BigDecimal) {
           return BigDecimalByteStringEncoder.encodeToBigNumericByteString(((BigDecimal) value));
-        } else if (value instanceof Double || value instanceof Float) {
```

Comment:
We still can have unsupported Number instance, is that right?

Reply:
I'm not sure what you mean. We accept all Number formats now, rather than only Double or Float. This enables us to accept integer values as input.

Reply:
I have reduced the scope of the change by simply adding Integer and Long alonside Double and Float.

Reply:
Sorry, I meant the conversion of Long to double value `BigDecimal.valueOf(((Number) value).doubleValue())` could lose precision here. Long has max 2^63-1 up to 18 digit of significance, however Double has 16 digit of significance. A safe way is something like this
```
else if (value instanceof Double || value instanceof Float) {
          return BigDecimalByteStringEncoder.encodeToNumericByteString(
              BigDecimal.valueOf(((Number) value).doubleValue()));
        }
ekse if (value instanceof java.lang.Short || value instanceof java.lang.Integer || value instanceof java.lang.Long) {
          return BigDecimalByteStringEncoder.encodeToNumericByteString(
              BigDecimal.valueOf(((Number) value).longValue()));
        }
```

my original comment about the unsupported Number refers to some known direct subclasses of Number here: https://docs.oracle.com/javase/8/docs/api/java/lang/Number.html

Reply:
I broke out handling of short/integer/long as you suggested. I have also revised the test case to include a value with 18 digits which demonstrates it could not be handled by a double.